mcs/class/corlib/System.Security.Permissions/SiteIdentityPermission.cs

```diff
@@ -197,7 +197,7 @@ private bool IsValid (string s)
 				if (x == 42) {
 					// special case for wildcards (*)
 					// must be alone or first and followed by a dot
-					if ((s.Length > 1) && ((s [i + 1] != '.') || (i > 0)))
```

Comment:
It took me a long time to understand the comment above.  So if there's a '*', the string must be either '*' or '*.' or else it's invalid?